src/acp/control.ts

```diff
@@ -167,34 +172,62 @@ export function getAccessControlAll(
- * @param accessControl The Access Control to insert into the given Access Control Resource.
- * @returns A new Access Control Resource equal to the given Access Control Resource, but with the given Access Control.
+ * @param withAccessControlResource A Resource with the Access Control Resource into which to insert an Access Control attached.
+ * @param accessControl The Access Control to insert into the Access Control Resource.
+ * @returns The given Resource with attached to it a new Access Control Resource equal to the original Access Control Resource, but with the given Access Control.
  */
-export function setAccessControl(
-  accessControlResource: AccessControlResource,
+export function setAccessControl<ResourceExt extends WithAccessibleAcr>(
+  withAccessControlResource: ResourceExt,
   accessControl: AccessControl
-): AccessControlResource {
-  return setThing(accessControlResource, accessControl);
+): ResourceExt {
+  const acr = getAcr(withAccessControlResource);
+  const updatedAcr = setThing(acr, accessControl);
+  const updatedResource = setAcr(withAccessControlResource, updatedAcr);
+  return updatedResource;
 }
 /**
  * ```{note} The Web Access Control specification is not yet finalised. As such, this
  * function is still experimental and subject to change, even in a non-major release.
  * ```
  *
- * Remove an [[AccessControl]] from an [[AccessControlResource]].
+ * Remove an [[AccessControl]] from an [[AccessControlResource]] attached to a Resource.
  *
- * @param accessControlResource The Access Control Resource to remove an Access Control from.
+ * @param withAccessControlResource A Resource with the Access Control Resource from which to remove an Access Control attached.
  * @param accessControl The Access Control to remove from the given Access Control Resource.
- * @returns A new Access Control Resource equal to the given Access Control Resource, excluding the given Access Control.
+ * @returns The given Resource with attached to it a new Access Control Resource equal to the original Access Control Resource, excluding the given Access Control.
  */
-export function removeAccessControl(
-  accessControlResource: AccessControlResource,
+export function removeAccessControl<ResourceExt extends WithAccessibleAcr>(
+  withAccessControlResource: ResourceExt,
   accessControl: AccessControl
-): AccessControlResource {
-  return removeThing(accessControlResource, accessControl);
+): ResourceExt {
+  const acr = getAcr(withAccessControlResource);
+  const updatedAcr = removeThing(acr, accessControl);
+  const updatedResource = setAcr(withAccessControlResource, updatedAcr);
+  return updatedResource;
+}
+
+function getAcr(resource: WithAccessibleAcr): AccessControlResource {
+  if (!hasAccessibleAcr(resource)) {
+    throw new Error(
+      "Cannot work with Access Controls on a Resource that does not have an Access Control Resource attached."
```

Comment:
Should we be trying to identify the Resource that does not have an ACR, i.e.:
```
"Cannot work with Access Controls on the Resource [${resource.toString()}] that does not have an Access Control Resource attached."
```
(or some meaningful equivalent of `.toString()` for a Resource)...?

Reply:
Yeah good idea: fed669c1543c4f8a21fa3c03ae9fe83d62064dff